pkg/controllers/jobset_controller.go

```diff
@@ -652,6 +660,13 @@ func setExclusiveAffinities(job *batchv1.Job, topologyKey string) {
 		})
 }
 
+func setNodeSelector(js *jobset.JobSet, job *batchv1.Job) {
+	if job.Spec.Template.Spec.NodeSelector == nil {
+		job.Spec.Template.Spec.NodeSelector = make(map[string]string)
+	}
+	job.Spec.Template.Spec.NodeSelector[jobset.JobSetNameKey] = js.Name
```

Comment:
Few things:
1) name alone is not unique, it needs to be namespaced
2) we need to inject the job name, not the jobset's name (each job lands on a node pool)
3) Lets also add a noschedule toleration

Reply:
Done